File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;

class Solution {
    public int[] solution(int brown, int yellow) {
        
        int answer[] = new int[2];
        
        for(int i=1; i<=(int)Math.sqrt(yellow); i++){       
            if(yellow%i != 0) continue;
            int width = yellow/i + 2;
            int height = i+2;
            int brownCnt = width*height - yellow;
            if(brownCnt == brown){
                answer[0] = width;
                answer[1] = height;
                break;
            } 
        }
     
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    
    for(let i=0; i<=Math.sqrt(yellow); i++){
        if(yellow%i != 0) continue;
        let width = yellow/i+2;
        let height = i+2;
        if(width*height-yellow == brown){
            answer.push(width);
            answer.push(height);
            break;
        }
    }
    
    return answer;
}